src/plugins.c

```diff
@@ -1516,36 +1639,44 @@ static void pm_show_dialog(GtkMenuItem *menuitem, gpointer user_data)
 
 	label = geany_wrap_label_new(_("Choose which plugins should be loaded at startup:"));
 
-	pm_widgets.keybindings_button = gtk_button_new_with_label(_("Keybindings"));
-	gtk_widget_set_sensitive(pm_widgets.keybindings_button, FALSE);
-	g_signal_connect(pm_widgets.keybindings_button, "clicked",
-		G_CALLBACK(pm_on_plugin_button_clicked), GINT_TO_POINTER(PM_BUTTON_KEYBINDINGS));
+	/* plugin popup menu */
+	pm_widgets.popup_menu = gtk_menu_new();
 
-	pm_widgets.configure_button = gtk_button_new_from_stock(GTK_STOCK_PREFERENCES);
-	gtk_widget_set_sensitive(pm_widgets.configure_button, FALSE);
-	g_signal_connect(pm_widgets.configure_button, "clicked",
-		G_CALLBACK(pm_on_plugin_button_clicked), GINT_TO_POINTER(PM_BUTTON_CONFIGURE));
+	menu_item = gtk_image_menu_item_new_from_stock(GTK_STOCK_PREFERENCES, NULL);
+	gtk_container_add(GTK_CONTAINER(pm_widgets.popup_menu), menu_item);
+	g_signal_connect(menu_item, "activate",
+			G_CALLBACK(pm_on_plugin_button_clicked), GINT_TO_POINTER(PM_BUTTON_CONFIGURE));
+	pm_widgets.popup_configure_menu_item = menu_item;
 
-	pm_widgets.help_button = gtk_button_new_from_stock(GTK_STOCK_HELP);
-	gtk_widget_set_sensitive(pm_widgets.help_button, FALSE);
-	g_signal_connect(pm_widgets.help_button, "clicked",
-		G_CALLBACK(pm_on_plugin_button_clicked), GINT_TO_POINTER(PM_BUTTON_HELP));
+	menu_item = gtk_image_menu_item_new_with_mnemonic(_("Keybindings"));
+	gtk_container_add(GTK_CONTAINER(pm_widgets.popup_menu), menu_item);
+	g_signal_connect(menu_item, "activate",
+			G_CALLBACK(pm_on_plugin_button_clicked), GINT_TO_POINTER(PM_BUTTON_KEYBINDINGS));
+	pm_widgets.popup_keybindings_menu_item = menu_item;
 
+	menu_item = gtk_image_menu_item_new_from_stock(GTK_STOCK_HELP, NULL);
+	gtk_container_add(GTK_CONTAINER(pm_widgets.popup_menu), menu_item);
+	g_signal_connect(menu_item, "activate",
+			G_CALLBACK(pm_on_plugin_button_clicked), GINT_TO_POINTER(PM_BUTTON_HELP));
+	pm_widgets.popup_help_menu_item = menu_item;
+
+	/* put it together */
 	hbox = gtk_hbox_new(FALSE, 0);
 	gtk_box_set_spacing(GTK_BOX(hbox), 6);
-	gtk_box_pack_end(GTK_BOX(hbox), pm_widgets.keybindings_button, FALSE, FALSE, 0);
-	gtk_box_pack_end(GTK_BOX(hbox), pm_widgets.configure_button, FALSE, FALSE, 0);
-	gtk_box_pack_end(GTK_BOX(hbox), pm_widgets.help_button, FALSE, FALSE, 0);
+	gtk_box_pack_start(GTK_BOX(hbox), filter_hbox, TRUE, TRUE, 0);
 
 	vbox2 = gtk_vbox_new(FALSE, 3);
 	gtk_box_pack_start(GTK_BOX(vbox2), label, FALSE, FALSE, 5);
+	gtk_box_pack_start(GTK_BOX(vbox2), hbox, FALSE, FALSE, 0);
 	gtk_box_pack_start(GTK_BOX(vbox2), swin, TRUE, TRUE, 0);
-	gtk_box_pack_start(GTK_BOX(vbox2), hbox, FALSE, TRUE, 0);
 
 	g_signal_connect(pm_widgets.dialog, "response", G_CALLBACK(pm_dialog_response), NULL);
 
 	gtk_box_pack_start(GTK_BOX(vbox), vbox2, TRUE, TRUE, 0);
 	gtk_widget_show_all(pm_widgets.dialog);
+	gtk_widget_show_all(pm_widgets.popup_menu);
+
+	gtk_widget_grab_focus(pm_widgets.tree);
```

Comment:
Just for the records: grabbing the focus was necessary to not leave the default focus on the filter entry but keep it on the plugin list as it was before.